src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -345,42 +347,53 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: DataSource,
           }.toSet
 
           if (allTheRefs.contains(UserService.allUsersGroupRef)) {
-            Future.successful(RequestComplete( ErrorReport(StatusCodes.BadRequest, s"Please contact an administrator to alter access to ${UserService.allUsersGroupRef.groupName}") ))
+            Future.successful(RequestComplete(ErrorReport(StatusCodes.BadRequest, s"Please contact an administrator to alter access to ${UserService.allUsersGroupRef.groupName}")))
+
+          } else if (!updateMap.get(Left(RawlsUser(userInfo))).forall(_ == containerDAO.authDAO.getMaximumAccessLevel(RawlsUser(userInfo), workspaceContext.workspace.workspaceId, txn))) {
```

Comment:
will comment. The extra case is if the user is in the updateMap but the value in the map is the same as their current value so there is not actually a change.